src/module/rent/controller/rent.controller.ts

```diff
@@ -0,0 +1,73 @@
+import { Application, Request, Response, NextFunction } from "express";
+import { RentService } from "../rent.module";
+import { CreateRentDto } from "../application/dto/create.rent.dto";
+import { fromRentDtoToEntity } from "../application/mapper/fromDtoRentToEntity";
+
+export class RentController {
+	baseRoute = "/rent";
+	private readonly rentService: RentService;
+	constructor(rentService: RentService) {
+		this.rentService = rentService;
+	}
+	configureRoutes(app: Application) {
+		app.post(`${this.baseRoute}`, this.create.bind(this));
+		app.get(`${this.baseRoute}`, this.getAll.bind(this));
+		app.get(`${this.baseRoute}/:id`, this.getById.bind(this));
+		app.put(`${this.baseRoute}/:id`, this.update.bind(this));
+		app.delete(`${this.baseRoute}/:id`, this.delete.bind(this));
+	}
+
+	async create(req: Request, res: Response, next: NextFunction) {
+		const newRent = new CreateRentDto(req.body);
+		try {
+			newRent.validate();
+			const rent = fromRentDtoToEntity(newRent);
+			const rentSaved = await this.rentService.create(rent);
```

Comment:
Your controller should not be responsible of mapping entities, it's single responsibility should be to handle the user request (ex: validate it). Let's refactor so your service receive a dto as a parameter that it could handle and map to a domain entity

Reply:
I was mistaken. I thought the controller was responsible for sending the required format to the service. The correction should be easy. Thx!